src/librustc_target/spec/mod.rs

```diff
@@ -247,11 +247,19 @@ macro_rules! supported_targets {
                         // run-time that the parser works correctly
                         t = Target::from_json(t.to_json())?;
                         debug!("Got builtin target: {:?}", t);
-                        Ok(t)
+                        return Ok(t)
                     },
                 )+
-                _ => Err(format!("Unable to find target: {}", target))
+                _ => {},
             }
+            let mut triple: Vec<&str> = target.splitn(3, '-').collect();
+            if triple.len() >= 3 {
+                if triple[1] != "unknown" {
+                    triple[1] = "unknown";
```

Comment:
Please add an explanation of what `triple[1]` represents and why this fallback makes sense. Right now it requires reading the PR description and linked issue to understand what is going on. Just from the code it isn't clear why there wouldn't also be a fallback for `triple[2]` being unknown.